c7n/resources/rdscluster.py

```diff
@@ -216,5 +211,141 @@ def process(self, snapshots):
         c = local_session(self.manager.session_factory).client('rds')
         for s in snapshots_set:
-            c.delete_db_cluster_snapshot(
-                DBClusterSnapshotIdentifier=s['DBClusterSnapshotIdentifier'])
+            try:
+                c.delete_db_cluster_snapshot(
+                    DBClusterSnapshotIdentifier=s['DBClusterSnapshotIdentifier'])
+            except ClientError as e:
+                raise
+
+
+@RDSClusterSnapshot.action_registry.register('restore')
+class RDSClusterSnapshotRestore(BaseAction):
+    schema = type_schema(
+        'restore', vpc_sec_groups={'type': 'array'}, port={'type': 'number'}, db_subnet_group_name={'type': 'string'},
+        snapshot_identifier={'type': 'string'})
+
+    def process(self, resources):
+
+        vpc_sec_groups = self.data.get('vpc-security-groups')
+        port = self.data.get('port')
+        db_subnet_group_name = self.data.get('db-security-group-name')
+        snapshot_id = self.data.get('snapshot-identifier');
+
+        for latest_snapshot in self.get_latest_snapshot(resources):
+            client = local_session(self.manager.session_factory).client('rds')
+
+            if vpc_sec_groups is None:
+                self.log_default_value('vpc_sec_group')
+            elif db_subnet_group_name is None:
+                self.log_default_value('db_subnet_group_name')
+            elif port is None:
+                self.log_default_value('port')
+            else:
+                log.info('USING PARAMS PROVIDED IN POLICY')
+
+            print 'USING SNAPSHOT FOR CLUSTER::', latest_snapshot['DBClusterIdentifier']
+            print 'USING SNAPSHOT::', latest_snapshot['DBClusterSnapshotIdentifier']
+
+            params = {'DBClusterIdentifier': latest_snapshot['DBClusterIdentifier'],
+                      'SnapshotIdentifier': latest_snapshot['DBClusterSnapshotIdentifier'],
+                      'Engine': latest_snapshot['Engine'],
+                      'Port': port,
+                      'VpcSecurityGroupIds': vpc_sec_groups,
+                      'DBSubnetGroupName': db_subnet_group_name
+                      }
+
+            if snapshot_id:
+                params['SnapshotIdentifier'] = snapshot_id
+
+            try:
+                client.restore_db_cluster_from_snapshot(**params)
+                self.create_db_instances_within_cluster(latest_snapshot)
+            except ClientError as e:
+                raise
+
+    """
+      Right now im only returning the first value returned by the cli. I sort them from bottom to top then append the top
+      value to the array then return the array.  I am setting this function up to handle an array of different
+      cluster's snapshot, but for right now I just want to get the single cluster restore working then Ill focus on
+      creating multiple rds cluster instances that have same values within an array in the policy.
+    """
+    @staticmethod
+    def get_latest_snapshot(snapshots):
+
```

Comment:
where is that enforced? an action ideally shouldn't assume extant filters, if it does then that needs to be explicit in the validate() method for the action.